Prog/Neighborhood change/4.1_prediction_OTR.R

```diff
@@ -0,0 +1,338 @@
+## predicting displacement risk with OTR based neighborhood change results
+## Yipeng Su
+## last updated 9/25/2024
+
+# Install pacman if not already installed
+if (!require("pacman")) install.packages("pacman")
+
+# Load the required packages using pacman
+pacman::p_load(
+  tidyverse, DescTools, purrr, tidycensus, mapview, stringr, educationdata, sf, 
+  readxl, urbnthemes, sp, ipumsr, survey, srvyr, dplyr, Hmisc, haven, caret, 
+  boot, corrplot, stargazer
+)
+
+census_api_key("05de4dca638d81abd2dc60d0d28e3781183e185e", install = TRUE)
+#get your key at https://api.census.gov/data/key_signup.html
+
+#update to your Box drive directory
+setwd("C:/Users/Ysu/Box/Greater DC/Projects/DMPED Housing Assessment 2024/Task 2 - Nbrhd Change and Displacement Risk Assessment/Data collection/")
+
+housingmarket <- read_csv("Clean/housingmarket.csv") 
+
+data <- read_sas("W:/Libraries/Realprop/Data/sales_sum_tr20.sas7bdat")
+#tring the OTR sales data for median home value
+
+OTR_sales <- data %>% 
+  rename(GEOID=Geo2020) %>% 
+  select(GEOID, mprice_tot_1999,mprice_tot_2000,mprice_tot_2001, mprice_tot_2012,mprice_tot_2011,mprice_tot_2010, mprice_tot_2013,mprice_tot_2022,mprice_tot_2021,mprice_tot_2023) %>% 
+  # filter(is.na(mprice_tot_2000)) #8 missing in 2022 #13 missing in 2012 #12 missing in 2000
+  mutate(GEOID=as.numeric(GEOID))
+
+#use the OTR data instead for home value
+housingmarket <- housingmarket %>% 
+  left_join(OTR_sales,by=c("GEOID")) %>% 
+  mutate(medianhome_2000_2020=mprice_tot_2000,
+         medianhome_2012_2020=mprice_tot_2012,
+         medianhome_2022=mprice_tot_2022) %>% 
+  # filter(is.na(medianhome_2022)|is.na(medianhome_2012_2020)) %>%
+  # select(GEOID, NBH_NAMES, total_hh_2022, medianhome_2000_2020,medianhome_2012_2020,medianhome_2022,mprice_tot_1999,mprice_tot_2001, mprice_tot_2011,mprice_tot_2021) %>%
+  mutate(medianhome_2000_2020=ifelse(GEOID=="11001004702",165000,medianhome_2000_2020),
```

Comment:
Same question as in 3.1. What are these hard-coded values?

Reply:
these are where OTR sales data is not available and I manually coded them when there is data available in previous year, if not the following year data

Reply:
Oh, right. If you would please note that in the code, that would be helpful.